src/creator.rs

```diff
@@ -77,12 +77,18 @@ impl<H: Hasher> Creator<H> {
 
         let new_preunit = PreUnit::new(self.node_ix, round, control_hash);
         debug!(target: "AlephBFT-creator", "{:?} Created a new unit {:?} at round {:?}.", self.node_ix, new_preunit, self.current_round);
-        self.new_units_tx
+        if let Err(e) = self
+            .new_units_tx
             .unbounded_send(NotificationOut::CreatedPreUnit(new_preunit))
-            .expect("Channel should be open");
+        {
+            debug!(target: "AlephBFT-creator", "{:?} notification channel is closed {:?}, closing", self.node_ix, e);
+            return Err(());
+        }
```

Comment:
For consistency, can we replace `if let` with  `.map_err(...)?` as in the extender

Reply:
Done.